### ISSUE TITLE
business case is not calculate

### ISSUE THREAD

**Issue by JeroenvdLogt** [quintel/etmoses#734]
I am trying to calculate a business case for: USEF Lochem traditioneel netverzwaring 2050 + marktmodel (exl flex)

But the BC is not updated by ET-Moses. I am sure I have transactions with government and producer but cannot find those in the BC table.

**Comment by ChaelKruip** [comment]
@grdw can you take a look?

**Comment by grdw** [comment]
It has to do with the merit price for electric vehicles. That profile + the merit order is momentarely being migrated as a preparation to the deploy (@antw could you verify this). This is why it's not working. In the mean time you can try the production environment of etmoses instead of the beta.

**Comment by JeroenvdLogt** [comment]
Unfortunetily I it doesn't work on the production server either. On the production server I cannot even make a LES with 
USEF Lochem 2050 berkeningen as topology and
USEF Lochem 2050 zonder flex as marktmodel.

**Comment by JeroenvdLogt** [comment]
Is this has something to do with the ETM release. If this is the case can you give an estimation how long this problem is sustained.

**Comment by grdw** [comment]
Hmm..  :disappointed: it seems that this is a different problem.

The *beta* problem has to do with the release.
The *production* problem has nothing to do with the release but with the error tagged in this issue.

The problem on production can be manually resolved. 

![screen shot 2016-02-02 at 12 06 18 pm](https://cloud.githubusercontent.com/assets/2676542/12747628/639157ee-c9a5-11e5-864e-d468498836e0.png)

Long story short: It has to do with this. You can't define two rules with the same set of stakeholders in a maket model. I think this is resolved but I'm not sure. ( @antw could you elaborate? )

**Comment by antw** [comment]
> You can't define two rules with the same set of stakeholders in a maket model. I think this is resolved but I'm not sure. ( @antw could you elaborate? )

This is a limitation of the Turbine library. You *can* have two (or more) payments between the same two stakeholders, but the measure must be unique in each case. You can't have "AC6" paying "system operator" for "max kW load" more than once.

It's a quite sensible limitation for Turbine, but it causes this problem with the market model. I intend to factor our Turbine as soon as the opportunity presents, but until then we have to work around the limitation by combining the two payment rules (delete the top one, and increase the tariff of the second to "6001.93").

**Comment by JeroenvdLogt** [comment]
Also with the adjusted market model I am not able to make my LES work... 
When going from ETM scaled down Lochem scenario to ET-Moses and chosing 
USEF Lochem 2050 berkeningen as topology and
USEF Lochem 2050 zonder flex as marktmodel

the technology table occurs but I am not able to do anything. It has crashed.

**Comment by JeroenvdLogt** [comment]
![image](https://cloud.githubusercontent.com/assets/7781037/12782121/616f4550-ca79-11e5-9cc1-7d9ac2b9d35d.png)

**Comment by ChaelKruip** [comment]
Adding priority label because @JeroenvdLogt cannot continue his work if this is not resolved.

**Comment by antw** [comment]
The duplicate payment rule shown in @grdw's screenshot above was still present in the market model. I have removed it and [a business case has now been calculated](http://moses.et-model.com/testing_grounds/395).

**Comment by ChaelKruip** [comment]
> I have removed it and a business case has now been calculated.

:+1:

**Comment by JeroenvdLogt** [comment]
![image](https://cloud.githubusercontent.com/assets/7781037/12782808/f0ca7a22-ca7d-11e5-8672-f73d9e7a1517.png)
It's a pity. This is an existing LES with an old topology (other yaml text same name) you show. I was talking about creating a new LES from ETM....

![image](https://cloud.githubusercontent.com/assets/7781037/12782696/3006908c-ca7d-11e5-9956-2b592a6b309d.png)

**Comment by antw** [comment]
This appears to be unrelated to the business case. Indeed the JavaScript for the technology table seems to hang, likely due to the very large topology. Perhaps, when time permits, @grdw can look into optimising this further and find out where the bottlenecks are (because the page is not *that* complex).

Unfortunately I cannot think of any quick work-arounds. Simplifying the topology should improve performance, but you shouldn't *have* to do that.

![screen shot 2016-02-03 at 13 05 35](https://cloud.githubusercontent.com/assets/4383/12783011/d6ffca04-ca76-11e5-8bf0-d359c935b102.png)

![screen shot 2016-02-03 at 13 05 59](https://cloud.githubusercontent.com/assets/4383/12783016/e276c496-ca76-11e5-84d5-c15d6db8f980.png)

**Comment by grdw** [comment]
I'm closing this issue. I've created a separate issue for the speed optimization. Will start whenever time permits.
